app/controllers/api/v1/journeys_controller.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class JourneysController < ApplicationController
+      skip_before_action :verify_authenticity_token
+
+      def create
+        card = find_card
+        unless card
+          render(json: { error: 'Card not found' }, status: :not_found)
+          return
+        end
+
+        journey = build_journey(card)
+        charge_for_journey(card, journey)
+
+        if journey.save
+          render json: journey.card, status: :created
+        else
+          render json: journey.errors, status: :unprocessable_entity
+        end
+      end
+
+      def update
+        card = find_card
+        unless card
+          render(json: { error: 'Card not found' }, status: :not_found)
+          return
+        end
```

Comment:
With the usage of `find` in `set_card` method you can get rid of this since `ApplicationController` will catch the error for you